core/templates/domain/objects/UnitsObjectFactory.ts

```diff
@@ -74,7 +74,10 @@ export class UnitsObjectFactory {
     for (var i = 0; i < units.length; i++) {
       if ('*/()# '.includes(units[i]) && unit !== 'per') {
         if (unit.length > 0) {
-          if ((unitList.length > 0) && this.isunit(unitList.slice(-1).pop())) {
+          const unitValue = unitList.slice(-1).pop();
+          if ((unitList.length > 0) &&
+            (typeof unitValue !== undefined) &&
+            this.isunit(<string>unitValue)) {
```

Comment:
Why is this needed? Also, probably worth separating into a function, so that the code is more readable.

Reply:
`pop()` will never return `undefined` since there is a check for `unitList` to not be empty. The compiler fails to make this conclusion so I used **type assertion** instead.